### PR TITLE
[Feature] Support primary key replicated storage

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 3 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/StreamLoadTask.java|2|3|66.67%|[262]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 3 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/StreamLoadTask.java|2|3|66.67%|[262]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 3 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/StreamLoadTask.java|2|3|66.67%|[262]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 3 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/StreamLoadTask.java|2|3|66.67%|[262]|

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by sevev — DISMISSED** [review]

**Comment by xiaoyong-z** [comment]
if replicated storage is turn on for primary table by default, shuffle service for stream load and broker load should be disabled

File: be/src/http/action/stream_load.cpp [hunk 1]
```diff
@@ -489,6 +489,15 @@ Status StreamLoadAction::_process_put(HttpRequest* http_req, StreamLoadContext*
             return Status::InvalidArgument("Invalid load_dop format");
         }
     }
+    if (!http_req->header(HTTP_ENABLE_REPLICATED_STORAGE).empty()) {
```
Comment (chaoyli) [inline be/src/http/action/stream_load.cpp:492]: How about the performance gained?

Comment (meegoo) [reply be/src/http/action/stream_load.cpp:492]: performance gained for primary key replicated storage?

Comment (chaoyli) [reply be/src/http/action/stream_load.cpp:492]: Yes. The performance gained between replicated storage and non-replicated storage for primary-key table

Comment (meegoo) [reply be/src/http/action/stream_load.cpp:492]: In theory, it is the same as other tables, and the specific results will be supplemented in the performance test of version 2.5.

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by meegoo** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 3 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/StreamLoadTask.java|2|3|66.67%|[262]|

**Comment by meegoo** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
run starrocks_admit_test

**Pull request merged**